src/utils/media-utils.js

```diff
@@ -544,3 +544,17 @@ export function closeExistingMediaMirror() {
     });
   }
 }
+
+export function hasAudioTracks(el) {
+  if (!el) return false;
+  // At some point browsers will implement the audioTracks property
+  if (el.audioTracks !== undefined) {
+    return el.audioTracks.length > 0;
+  } else if (el.webkitAudioDecodedByteCount !== undefined) {
+    return el.webkitAudioDecodedByteCount > 0;
```

Comment:
Might this be 0 when the video is first loading? We wait in `createVideoTextureAudioSourceEl` for the video to have a width/height so I guess at that point we could probably also assume some audio should have been decoded.

Reply:
I'd say so, for what I see about media element loaddata event https://html.spec.whatwg.org/multipage/media.html#event-media-loadedmetadata that's when the dimensions are known for the whole media resource (complete set of media data). We are checking for the actual dimensions instead of the loaddata event (I don't why is that though) so I think it's safe to assume that all tracks are available by that time and we can rely on that instead of adding overhead to the tick.